plugin/plugin.go

```diff
@@ -0,0 +1,214 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/PagerDuty/go-pagerduty"
+	"github.com/sirupsen/logrus"
+)
+
+// Severity levels as constants for reuse and readability.
+const (
+	SeverityCritical = "critical"
+	SeverityError    = "error"
+	SeverityWarning  = "warning"
+	SeverityInfo     = "info"
+)
+
+// Args provides plugin execution arguments.
+type Args struct {
+	Level             string `envconfig:"PLUGIN_LOG_LEVEL"`
+	RoutingKey        string `envconfig:"PLUGIN_ROUTING_KEY"`
+	IncidentSummary   string `envconfig:"PLUGIN_INCIDENT_SUMMARY"`
+	IncidentSource    string `envconfig:"PLUGIN_INCIDENT_SOURCE"`
+	IncidentSeverity  string `envconfig:"PLUGIN_INCIDENT_SEVERITY"`
+	DedupKey          string `envconfig:"PLUGIN_DEDUP_KEY"`
+	CreateChangeEvent bool   `envconfig:"PLUGIN_CREATE_CHANGE_EVENT"`
+	Resolve           bool   `envconfig:"PLUGIN_RESOLVE"`
```

Comment:
```suggestion
	ResolveIncident   bool   `envconfig:"PLUGIN_RESOLVE_INCIDENT"`
```

let's update it for clarity